### DISCUSSION TITLE
Changes made to openstreetmap.org, how to see them in Cruiser?

### DISCUSSION THREAD

**Discussion by ramondevic** [devemux86/cruiser#422] [Q&A] [answered]
Every now and then I go on a drive and Cruiser (routing with GraphHopper) suggests I go through a place where it's really forbidden.
I have notified openstreetmap.org that the street is indeed prohibited for motor vehicles and I assume that in the next few days it will be modified.

The question is: for Cruiser to take this into account, do I need to download the sitemap or download the GraphHopper routing files?

(translated from Catalan)

**Comment by devemux86** [comment]
@ramondevic thanks for the question.

OpenStreetMap data is used in 2 things: display maps and calculate routes.

- OpenStreetMap changes appear on the map after we update the maps.

- OpenStreetMap changes appear in route calculations after we update the routing data for:
  - BRouter
  - GraphHopper

In your case, update GraphHopper data when it's available in Cruiser, i.e. after server rebuilds your region.

Note: BRouter data is updated daily.

**Reply by ramondevic** [reply]
According to the data from https://download.mapsforge.org/ it seems that the maps and routes have not been updated for a long time ?

https://download.mapsforge.org/maps/v5/europe/
![imatge](https://github.com/devemux86/cruiser/assets/122691401/55f26555-8edc-4c20-8867-793faa8cfb0a)

https://download.mapsforge.org/graphs/europe/
![imatge](https://github.com/devemux86/cruiser/assets/122691401/14d46939-77f6-4b5c-8dcd-71c305934f3f)

**Reply by devemux86** [reply]
I'll take a look. Maps could delay the process due to their complexity.
POI and routing are much simpler, but everything is created together.

**Reply by airald** [reply]
I think that maps are updated each "many months", I remember a post (not sure here) talking about it. BRouter is daily updated, I've made many changes and I see the effects one day after.

**Reply by devemux86** [reply]
Indeed, as maps usually do not need frequent updates.
Routing can be more up to date and BRouter has daily updates.

POI: if we really want today's results, we can use just the online search.

**Reply by ramondevic** [reply]
Are BRouter changes made on openstreetmap.org?
If so and BRouter is updated daily, couldn't GraphHopper routes be updated more often?

The maps don't change that much either, but once a month or every two months it would be nice if they were updated.

**Reply by devemux86** [reply]
@ramondevic BRouter also uses OpenStreetMap data.

**Reply by ramondevic** [reply]
I've recently made some changes to OpenStreetMap of streets that are now off limits and other roads that no longer exist.

At https://www.openstreetmap.org the changes are already visible but at https://download.mapsforge.org/graphs/ they are not available, in fact most of the data is from October-2023.

How can the download files be updated?

Thanks.

**Reply by devemux86** [reply, marked as answer]
@ramondevic 

Server creates maps + poi + graphs for all countries, so it can take a while to complete a whole update.

There are already updated maps (2024-01) for some continents, Europe seems to be next.

For instant updates, better try BRouter (offline) as it is updated daily.

**Reply by MotoUKRider** [reply]
I did some [similar editing with a local bridge toll](https://github.com/devemux86/cruiser/discussions/134#discussioncomment-7380651) and made use of BRouter's very quick update the next day.  Naturally, my digging around highlighted an issue with BRouter not handling the "rules" of the vehicle+toll correctly :laughing:

That situation (detailed in my link) was one of those unique/local things that users who have more than a passing interest in "fun" routing will find: A specific road, bridge or access-way that you know is wrong on the map.  Mostly it's quite easy to fix it and wait for the update but when you stumble-upon a software limitation then it's quite nice knowing you're helping the product.

Lots of map resources build quarterly, and when you see the CPU crunching involved you understand why more people don't do it.  I've had success building the "Isle of Man" map for BaseCamp (like the old Dutch on-demand server) with a batch file and a corn-maze of command-line options... it works, but I wouldn't fancy doing all of Europe!

**Comment by ramondevic** [comment]
Okay, so we'll have to wait :(

In any case, if Graphs are created daily in BRouter, why can't the same be done with GraphHopper (even if it was only once a week)?

**Reply by devemux86** [reply]
BRouter only creates its own routing data.
Mapsforge server creates maps + poi + graphs.

You can use BRouter, it has detailed routing options. :slightly_smiling_face:

**Reply by ramondevic** [reply]
That's it, it was an improvement that I thought could be interesting for users who want to calculate the route from the phone, I almost always design the route at "https://kurviger.de/" and then export it with ". kurviger" and it works great.

I only design the route on the phone very few times and I'm especially interested that when I navigate I can see on the phone:
. destination (next waypouint)
. the next street/road I will reach
. the place I'm at right now
... and that doesn't work with BRouter and that's why I use GraphHopper

**Reply by ramondevic** [reply]
![imatge](https://github.com/devemux86/cruiser/assets/122691401/dd840a6b-1706-414b-9043-aad64930bdc3)

**Reply by devemux86** [reply]
@ramondevic

If you use kurviger routes then the street names are always available in navigation.
(they are provided by the route)

Do you want to use BRouter (offline) or GraphHopper (offline) for offline rerouting?
BRouter does not contain street names, but only in the part of the rerouting.
